MediaBrowser.Controller/MediaEncoding/EncodingHelper.cs

```diff
@@ -2226,45 +2226,22 @@ public string GetAudioFilterParam(EncodingJobInfo state, EncodingOptions encodin
 
             if (isTranscodingAudio)
             {
-                // Set max transcoding channels to 8 to prevent encoding errors due to asking for too many channels
+                // Set max transcoding channels for encoders that can't handle more than a set amount of channels
                 // AAC, FLAC, ALAC, libopus, libvorbis encoders all support at least 8 channels
-                int transcoderChannelLimit = 8;
+                int transcoderChannelLimit;
 
-                // Set max transcoding channels for encoders that can't handle more than a set amount of channels
-                if (codec.IndexOf("wma", StringComparison.OrdinalIgnoreCase) != -1)
-                {
-                    // wmav2 currently only supports two channel output
-                    transcoderChannelLimit = 2;
-                }
-                else if (codec.IndexOf("mp3", StringComparison.OrdinalIgnoreCase) != -1)
-                {
-                    // libmp3lame currently only supports two channel output
-                    transcoderChannelLimit = 2;
-                }
-                else if (codec.IndexOf("ac3", StringComparison.OrdinalIgnoreCase) != -1)
-                {
-                    // ac3 encoder currently only supports six channel output
-                    transcoderChannelLimit = 6;
-                }
-                else if (codec.IndexOf("eac3", StringComparison.OrdinalIgnoreCase) != -1)
-                {
-                    // eac3 encoder currently only supports six channel output
-                    transcoderChannelLimit = 6;
-                }
-                else if (codec.IndexOf("dts", StringComparison.OrdinalIgnoreCase) != -1)
-                {
-                    // dca encoder currently only supports six channel output
-                    transcoderChannelLimit = 6;
-                }
-                else if (codec.IndexOf("mlp", StringComparison.OrdinalIgnoreCase) != -1)
-                {
-                    // mlp encoder currently only supports six channel output
-                    transcoderChannelLimit = 6;
-                }
-                else if (codec.IndexOf("truehd", StringComparison.OrdinalIgnoreCase) != -1)
+                switch (GetAudioEncoder(state))
```

Comment:
Don't use switch statement to compare strings, it doesn't behave the same as before

Reply:
Is this still the case even though output of GetAudioEncoder() has ToLowerInvariant called if not otherwise defined?

Reply:
I would probably work, but just to be consistent with the rest of the codebase I'd use the string comparison functions

Reply:
I've switched to using string comparison. Let me test to make sure this is working. Is the switch expression I have now okay or would it be better to make into if statements instead?

Reply:
I've changed to string comparison functions in a switch expression. If it would be better to do if statements instead, let me know.